### PR TITLE
Pedersen hash

### PR DESCRIPTION
**Pull request by pefontana**
# Pedersen hash
* Create the StarkHash trait
* Create the Pedersen struct that implements the StarkHash trait
* Update README.md

Please add the labels corresponding to the type of changes your PR introduces:
- Feature

## Does this introduce a breaking change?

No

### CODE REVIEW

**Review by LucasLvy — COMMENTED** [review]
I think we should have a hash trait so that we can swap between poseidon and pedersen easily in libs/projects

**Comment by pefontana** [comment]
> I think we should have a hash trait so that we can swap between poseidon and pedersen easily in libs/projects

Sure @LucasLvy , sounds good
I created the StarkHash trait
https://github.com/starknet-io/types-rs/pull/12/files#diff-e595c2e9b33ab8cfece5d30cb912859fbf7aba4fbb84bb2895cbb168dc0b620e

File: crates/starknet-types-core/README.md [hunk 1]
```diff
@@ -6,12 +6,26 @@ Core types representation for Starknet.
 
 The `starknet-types-core` crate provides:
 * The universal `Felt` (Field Element) type for Cairo and STARK proofs. It was created to reduce the fragmentation in the Starknet Rust ecosystem by providing a standardized representation of the `Felt` type.
-* The `AffinePoint` and `ProjectivePoint` structs, which represent points on the Stark curve for performing elliptic curve operations.
 
 ## Features
 
-- Standardized `Felt` type: Simplify your codebase by using our standardized `Felt` type.
-- Optimized for performance: The `Felt` type has been optimized for high-performance applications.
+### No features
```
Comment (Oppen) [inline crates/starknet-types-core/README.md:12]: "Always-on" sounds more appropriate.

Comment (Oppen) [reply crates/starknet-types-core/README.md:12]: Alternatively something like "baseline".

File: crates/starknet-types-core/src/hash/pedersen.rs [hunk 2]
```diff
@@ -0,0 +1,68 @@
...
+        Felt(hash)
+    }
+
+    /// Computes the Pedersen hash of an array of Felts, as defined
+    /// in <https://docs.starknet.io/documentation/architecture_and_concepts/Hashing/hash-functions/#array_hashing.>
+    fn hash_array(felts: &[Felt]) -> Felt {
+        let pedersen = PedersenLambdaworks::default();
+        let data_len =
+            Felt::from_u128(u128::try_from(felts.len()).expect("Got 2^128 felts or more."))
+                .unwrap();
```
Comment (Oppen) [inline crates/starknet-types-core/src/hash/pedersen.rs:29]: You can use the infallible `from` here:
```suggestion
            Felt::from(felts.len());
```

**Review by juanbono — APPROVED** [review]

**Comment by LucasLvy** [comment]
plz fix what @Oppen said so we can merge and get rolliiiiiiinnnnnn

File: crates/starknet-types-core/README.md (hunk 1 above)
Comment (pefontana) [reply crates/starknet-types-core/README.md]: Great!
done [80154b5](https://github.com/starknet-io/types-rs/pull/12/commits/80154b53c590defef75dbde3f50d21fd6a74f85d)

File: crates/starknet-types-core/src/hash/pedersen.rs (hunk 2 above)
Comment (pefontana) [reply crates/starknet-types-core/src/hash/pedersen.rs]: Thanks!, nice one
Done [3363c1b](https://github.com/starknet-io/types-rs/pull/12/commits/3363c1bcd55f3bb30418ea6bf1d9385f45e23944)

**Review by Oppen — APPROVED** [review]

**Pull request merged**